### ISSUE TITLE
ArgumentException: Error status code must not be OK

### ISSUE THREAD

**Issue by tobiasjacobsen** [grpc/grpc-dotnet#1870]
We're getting a lot of these errors lately from namespace: Grpc.Net.Client.Internal.GrpcCall. 
We're on latest version and running .NET 6 linux in Docker.

Do you guys have any idea what it could be?

**Details**

Error message: Error starting gRPC call.

```csharp
System.ArgumentException: Error status code must not be OK. (Parameter 'status')
  ?, in PickResult PickResult.ForFailure(Status status)
  ?, in PickResult ErrorPicker.Pick(PickContext context)
  ?, in async ValueTask<(Subchannel Subchannel, BalancerAddress Address, ISubchannelCallTracker SubchannelCallTracker)> ConnectionManager.PickAsync(PickContext context, bool waitForReady, CancellationToken cancellationToken)
  ?, in async Task<HttpResponseMessage> BalancerHttpHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
  ?, in async Task GrpcCall<TRequest, TResponse>.RunCall(HttpRequestMessage request, TimeSpan? timeout)
```

**Comment by JamesNK** [comment]
I think you've found a couple of bugs in the gRPC client. I have a PR to fix this in the next release (2.49.0).

PR: https://github.com/grpc/grpc-dotnet/pull/1872

Downgrading Grpc.Net.Client to 2.47.0 should fix this issue.

**Comment by tobiasjacobsen** [comment]
Thanks for the quick turnaround!
